File: 1.x/src/com/relevantcodes/extentreports/support/RegexMatcher.java
/*
* Copyright (c) 2015, Anshoo Arora (Relevant Codes).  All rights reserved.
* 
* Copyrights licensed under the New BSD License.
* 
* See the accompanying LICENSE file for terms.
*/


package com.relevantcodes.extentreports.support;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class RegexMatcher {
	
	public static Matcher getMatch(String line, String pattern) {
	      Pattern p = Pattern.compile(pattern);
	      Matcher m = p.matcher(line);
	      
	      if (m.find()) {
	    	  return m;
	      }
	      
	      return null;
	}
	
	public static String getNthMatch(String line, String pattern, Integer n) {
		try {
			return getMatch(line, pattern).group(n);
		}
		catch (Exception e) {
			e.printStackTrace();
			return null;
		}
	}
}

File: dotnet/ExtentReports/Support/RegexMatcher.cs
// ***********************************************************************
// Copyright (c) 2015, Anshoo Arora (Relevant Codes). All rights reserved.
//
// Copyrights licensed under the New BSD License.
//
// See the accompanying LICENSE file for terms.
// ***********************************************************************

namespace RelevantCodes.ExtentReports.Support
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Text.RegularExpressions;

    internal class RegexMatcher
    {
        public static string GetNthMatch(string Text, string Pattern, int MatchNumber)
        {
            var groups = GetMatch(Text, Pattern);

            if (groups != null)
            {
                return groups[MatchNumber].ToString();
            }

            return null;
        }

        public static GroupCollection GetMatch(string Text, string Pattern)
        {
            Match match = Regex.Match(Text, Pattern);

            if (match.Success)
            {
                return match.Groups;
            }

            return null;
        }
    }
}
